src/nupic/algorithms/ColumnPooler.cpp

```diff
@@ -0,0 +1,565 @@
+ 
+    Real stability_rate,
+    Real fatigue_rate,
+
+    Real period,
+    Int  seed,
+    bool verbose) {
+      initialize(
+       proximalInputDimensions, 
+	distalInputDimensions,
+	inhibitionDimensions,
+	cellsPerInhibitionArea,
+
+	sparsity,
+
+	potentialPool,
+	proximalSegments,
+	proximalSegmentThreshold,
+	proximalIncrement,
+	proximalDecrement,
+	proximalSynapseThreshold,
+
+	distalMaxSegments,
+	distalMaxSynapsesPerSegment,
+	distalSegmentThreshold,
+	distalSegmentMatch,
+       distalAddSynapses,
+	distalIncrement,
+	distalDecrement,
+	distalMispredictDecrement,
+	distalSynapseThreshold,
+
+	stability_rate,
+	fatigue_rate,
+
+	period,
+	seed,
+	verbose);
+  }
+
+
+  void initialize(
+        const vector<UInt> proximalInputDimensions,
+        const vector<UInt> distalInputDimensions,
+        const vector<UInt> inhibitionDimensions,
+        UInt               cellsPerInhibitionArea,
+
+        Real sparsity,
+
+        Topology_t potentialPool,
+        UInt       proximalSegments,
+        UInt       proximalSegmentThreshold,
+        Permanence proximalIncrement,
+        Permanence proximalDecrement,
+        Permanence proximalSynapseThreshold,
+
+        UInt       distalMaxSegments,
+        UInt       distalMaxSynapsesPerSegment,
+        UInt       distalSegmentThreshold,
+        UInt       distalSegmentMatch,
+        UInt       distalAddSynapses,
+        Permanence distalIncrement,
+        Permanence distalDecrement,
+        Permanence distalMispredictDecrement,
+        Permanence distalSynapseThreshold,
+
+        Real stability_rate,
+        Real fatigue_rate,
+
+        Real period,
+        Int  seed,
+        bool verbose) {
+
+    proximalInputDimensions_ = proximalInputDimensions;
+    distalInputDimensions_   = distalInputDimensions;
+    inhibitionDimensions_    = inhibitionDimensions;
+    cellsPerInhibitionArea_  = cellsPerInhibitionArea;
+
+    proximalSegments_        = proximalSegments;
+    this->sparsity                    = sparsity;
+    this->proximalSegmentThreshold    = proximalSegmentThreshold;
+    this->proximalIncrement           = proximalIncrement;
+    this->proximalDecrement           = proximalDecrement;
+    this->proximalSynapseThreshold    = proximalSynapseThreshold;
+    this->stability_rate              = stability_rate;
+    this->fatigue_rate                = fatigue_rate;
+
+    SDR proximalInputs(  proximalInputDimensions );
+    SDR inhibitionAreas( inhibitionDimensions );
+    cellDimensions_ = inhibitionAreas.dimensions;
+    cellDimensions_.push_back( cellsPerInhibitionArea );
+    SDR cells( cellDimensions_ );
+    rng_ = Random(seed);
+
+    // Setup the proximal segments & synapses.
+    proximalConnections.initialize(cells.size, proximalSynapseThreshold);
+    SDR_Sparsity PP_Sp(            proximalInputs.dimensions, 10 * proximalInputs.size);
+    SDR_ActivationFrequency PP_AF( proximalInputs.dimensions, 10 * proximalInputs.size);
```

Comment:
are these statistics really needed? 
I suggest a few options: 
- remove completely, let that on the user
- #ifdef DEBUG
- if(verbose > 1)
- make PP_Sp, PP_AF optional params of the constructor, if nullptr, skip these calls